tools/background_removal/background_removal.py

```diff
@@ -0,0 +1,43 @@
+import argparse
+import warnings
+
+
+import skimage.io
+from skimage.filters import difference_of_gaussians
+from skimage.io import imread
+from skimage.morphology import disk, white_tophat
+from skimage.restoration import rolling_ball
+
+
+def process_image(args):
+    image = imread(args.input_image)
+
+    if args.filter == "rolling_ball":
+        background_rolling = rolling_ball(image, radius=args.radius)
+        output_image = image - background_rolling
+
+    elif args.filter == "dog":
+        output_image = difference_of_gaussians(image, low_sigma=0, high_sigma=args.radius)
+
+    elif args.filter == "top_hat":
+        output_image = white_tophat(image, disk(args.radius))
+
+    with warnings.catch_warnings():
+        output_image = skimage.util.img_as_uint(output_image)
```

Comment:
```suggestion
        output_image = convert_image_to_format_of(output_image, image)
```
and then add a function like:
```python
def convert_image_to_format_of(image, format_image):
    """
    Converts the first image to the format of the second image.
    """
    if format_image.dtype == image.dtype:
        return image
    elif format_image.dtype == np.uint8:
        return skimage.util.img_as_ubyte(image)
    elif format_image.dtype == np.uint16:
        return skimage.util.img_as_uint(image)
    elif format_image.dtype == np.int16:
        return skimage.util.img_as_int(image)
    else:
        raise ValueError(f'Unsupported image data type: {format_image.dtype}')
```